src/osd/PrimaryLogPG.cc

```diff
@@ -5700,9 +5702,14 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	    oi.truncate_size = op.extent.truncate_size;
 	  }
 	}
-	result = check_offset_and_length(op.extent.offset, op.extent.length, cct->_conf->osd_max_object_size);
-	if (result < 0)
+	result = check_offset_and_length(op.extent.offset, op.extent.length,
+          osd_max_object_size);
+	if (result < 0) {
+          derr << __func__
+            << " osd_max_object_size >= 4GB; BlueStore has hard limit of 4GB."
```

Comment:
We shouldn't reference BlueStore or a constant hard limit in the OSD error code (they might be using FileStore!)
Just say it exceeds the max size.